src/server/job_func.c

```diff
@@ -792,7 +792,8 @@ job_purge(job *pjob)
 				update_subjob_state(pjob, JOB_STATE_EXPIRED);
 		}
 
-		(void)account_entity_limit_usages(pjob, NULL, NULL, DECR);
+		(void)account_entity_limit_usages(pjob, NULL, NULL, DECR,
+				pjob->ji_etlimit_decr_queued ? ETLIM_ACC_ALL_MAX : ETLIM_ACC_ALL);
```

Comment:
if `pjob->ji_etlimit_decr_queued` is set, then we now have to `DECR` only `set_entity_*_sum_max` as `set_entity_*_sum_queued` was already decremented when the job entered run. Hence passing `ETLIM_ACC_ALL_MAX`

if `pjob->ji_etlimit_decr_queued` is unset, then we have to decrement all the 4 limit attributes. Hence passing `ETLIM_ACC_ALL`